azuredevops/azuredevops.go

```diff
@@ -17,69 +17,98 @@ const (
 	userAgent = "go-azuredevops"
 )
 
-// Client for interacting with the Azure DevOps API
-type Client struct {
+// DevOpsClient represents methods for interacting with Core level of the Azure DevOps API
+// More info: https://docs.microsoft.com/en-us/rest/api/azure/devops/core/
+type DevOpsClient struct {
 	client *http.Client
 
 	BaseURL   string
 	UserAgent string
 
 	Account   string
-	Project   string
 	AuthToken string
 
+	// Services used to proxy to other API endpoints
+	Favourites *FavouritesService
+	Projects   *ProjectsService
+	Teams      *TeamsService
+}
+
+// ProjectClient represents methods for interacting with Project level of the Azure DevOps API
+type ProjectClient struct {
+	devOpsClient *DevOpsClient
+
+	Project string
+
 	// Services used to proxy to other API endpoints
 	Boards           *BoardsService
 	BuildDefinitions *BuildDefinitionsService
 	Builds           *BuildsService
 	DeliveryPlans    *DeliveryPlansService
-	Favourites       *FavouritesService
 	Git              *GitService
 	Iterations       *IterationsService
 	PullRequests     *PullRequestsService
-	Teams            *TeamsService
 	Tests            *TestsService
 	WorkItems        *WorkItemsService
 }
 
-// NewClient gets a new Azure DevOps Client
-func NewClient(account string, project string, token string) *Client {
-	c := &Client{
+// NewClient gets a new Project Client for Azure DevOps API
+func NewClient(account string, project string, token string) *ProjectClient {
```

Comment:
Can this be renamed to `NewProjectClient` please? There are breaking changes throughout this PR, so to keep `NewClient` as is, doesn't really add value. I'd prefer to be explicit please.